lib/Dialect/TritonGPU/IR/LinearLayoutConversions.cpp

```diff
@@ -321,16 +323,39 @@ LinearLayout sharedToLinearLayoutLeadingOffset(ArrayRef<int64_t> shape,
       bases2D.push_back({row, vec * ((row / perPhase) % maxPhase)});
     }
   }
-  LinearLayout tileLayout =
-      LinearLayout({{S("offset"), bases2D}}, {rowDimName, colDimName});
 
-  // Add the remaining dimensions.
-  for (int dim = batchDims - 1; dim >= 0; --dim) {
-    tileLayout *= LinearLayout::identity1D(shapePerCTA[dim], S("offset"),
-                                           outDimNames[dim]);
+  // Then distribute the remaining rows.
+  for (int logRow = llvm::Log2_32(tileRows);
+       logRow < llvm::Log2_32(collapsedShapePerCTA[0]); logRow++) {
+    bases2D.push_back({1 << logRow, 0});
+  }
+
+  auto outDimNames = standardOutDimNames(ctx, 2);
+  std::reverse(outDimNames.begin(), outDimNames.end());
+  LinearLayout tileLayout = LinearLayout({{S("offset"), bases2D}}, outDimNames);
```

Comment:
I'm trying to understand this code, why do you reverse the out dim names here? This means that `dim1` with the size `collapsedShapePerCTA[0]` and vice versa which doesn't seem to be how it was done before. Is this something to do with how the reshaped dimensions are collapsed?

Reply:
The test fails if I remove the reverse so it does seem to be load bearing.